### DISCUSSION TITLE
Does whisper has any method for transcribing Audio in base64 format?

### DISCUSSION THREAD

**Discussion by gupta9ankit5** [openai/whisper#1653] [Q&A]
I've hosted the Whisper model as a REST endpoint for transcribing audio files. However, due to the limitations of sending audio files in a JSON request, I had to encode them into base64 format before sending them to the endpoint. On the server-side, I'm currently decoding these base64-encoded audio files, regenerating them in the WAV format, and then uploading them to disk. Finally, I'm transcribing all of these uploaded audio files using Whisper's transcribe() method.

I'm wondering if there's a more efficient way to avoid the need for this regeneration and file upload process?

**Comment by jongwook** [comment]
You might want to make the endpoint to accept a binary audio file as the entire body of the POST message or as a multipart FORM field. [The Whisper API](https://platform.openai.com/docs/guides/speech-to-text) uses the latter option.
